### DISCUSSION TITLE
Adding the option to set / override the remaining items for custom recipe choices

### DISCUSSION THREAD

**Discussion by aerulion** [PaperMC/Paper#10573] [Ideas]
### Is your feature request related to a problem?

I would love the be easily able to modify the "remaining items" of custom recipes. Similar to a milk bucket resulting in an empty bucket when taking the item out the crafting window, I would like to define a custom item stack remaining for custom ingredients.

### Describe the solution you'd like.

I'm not quite sure what would be the best way to introduce that feature, ideally one would be able to define the remaining item of a RecipeChoice/ExactChoice, although that might be a bit misleading since it only applies to crafting recipes, and not for example something like furnace recipes (although maybe possible, but all vanilla items with remaining items are only part of crafting recipes).

### Describe alternatives you've considered.

Listening to the CraftItemEvent, manually checking my custom recipes and adding back the items I want to "return" in the appropriate slots.

If I missed any easier / proper way to achieve said functionality please let me know.

**Comment by Andre601** [comment]
Just want to mention something related, which is, that I proposed a way to set remaining items for the CrafterCraftEvent to Spigot, which got accepted and resulted in a `getRemainingItems()` method returning a `List<ItemStack>` that could be modified.

While that method is useful for the Crafter - and should probs be ported to Paper too - would I go a different route with the CraftItemEvent, since here, the items would remain in the crafting grid, while the Crafter spits them out alongside the item.
So it would need a way for us to define the Slot in the crafting grid to put the item at.
Tho, the event also extends InventoryClickEvent, so you could just set the inventory content per slot too I would assume...

Either way, point is, a method to get the ItemStacks returned in the crafting grid would be useful for sure.
